### PR TITLE
Testing

### LINKED ISSUE #112: Testing: apiCalls - getAllProviders() tests
For the fetch call `getAllProviders` within the `utils/apiCalls/DoctorsAndProviders` directory, 
- [ ] There should be a describe block
- [ ] A beforeEach(()) with the necessary information for the fetch call (id, mock response, URL)

**There should be three tests:**
- [ ] Should call fetch with the correct URL
- [ ] Should return the data we are expecting
- [ ] Should return an error if the ok property in the Promise object is false

### LINKED ISSUE #113: Testing: apiCalls - getDoctorsByLocation()
For the fetch call `getDoctorsByLocation` within the `utils/apiCalls/DoctorsAndProviders` directory, 
- [ ] There should be a describe block
- [ ] A beforeEach(()) with the necessary information for the fetch call (id/state code/etc, mock response, URL)

**There should be three tests:**
- [ ] Should call fetch with the correct URL
- [ ] Should return the data we are expecting
- [ ] Should return an error if the ok property in the Promise object is false

### LINKED ISSUE #114: Testing: apiCalls - getDoctorsForProviderByLocation()
For the fetch call `getDoctorsForProviderByLocation` within the `utils/apiCalls/DoctorsAndProviders` directory, 
- [ ] There should be a describe block
- [ ] A beforeEach(()) with the necessary information for the fetch call (id, mock response, URL)

**There should be three tests:**
- [ ] Should call fetch with the correct URL
- [ ] Should return the data we are expecting
- [ ] Should return an error if the ok property in the Promise object is false

### PR DESCRIPTION
**Pull request by nathangthomas**
## What does this PR do?
 This PR adds tests for the following: 
     - `getDoctorsByLocation( )`
     - `getAllProviders( )`
     - `getDoctorsForProviderByLocation( )`
     - adds sad path test for a diagnosis id that does not exists in `getAllProviders( )`

### Where should the reviewer start?
 Look at the test files named: 
     - 'utils/apiCalls/Conditions/getConditionById.test.js'
     - 'utils/apiCalls/DoctorsAndProviders/getAllProviders.test.js'
     - 'utils/apiCalls/DoctorsAndProviders/getDoctorsForProviderByLocation.test.js'
     - 'utils/apiCalls/DoctorsAndProviders/getDoctorsByLocation.test.js'

#### How should this be manually tested?
     - run the test suite with `npm test` or a single test with `npm test -- <file_path>`

#### Have tests been written for all functionality?
     -  I am still working on testing for `sendEmailReport( )`

#### Any background context you want to provide?
     n/a

#### What are the relevant tickets? 
Closes #112 
Closes #113 
Closes #114 

#### Additional Notes
     - I am still working on testing for `sendEmailReport( )` I will try to have a PR up tomorrow for ticket #115

### CODE REVIEW

File: utils/apiCalls/Conditions/getConditionById.test.js [hunk 1]
```diff
@@ -54,4 +54,8 @@ describe('getConditionById', () => {
       )
     );
   });
+
+  it('should return an error if getConditionById is passed an improper id', () => {
```
Comment (vrandall66) [inline utils/apiCalls/Conditions/getConditionById.test.js:58]: Nice!

File: utils/apiCalls/DoctorsAndProviders/getAllProviders.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      });
+    });
+    expect(getAllProviders()).rejects.toEqual(
+      Error(
+        'Could not get providers, please try again later.'
+      )
+    );
+  });
+
+});
```
Comment (vrandall66) [inline utils/apiCalls/DoctorsAndProviders/getAllProviders.test.js:46]: This all looks great, well done!

**Review by vrandall66 — APPROVED** [review]
This all looks great to me! Thank you so much for your help on this! I'll let you know if I run into any issues for whatever reason.

Also, if you edit the body of the PR and change the ticket format to:
Closes #actual ticket number here
Closes #another real number
etc...
it will automatically move those tickets/issues to the done column on the project board! Super neat :nerd_face:

**Comment by nathangthomas** [comment]
Ah, thanks for the tip. I actually like the satisfaction of dragging them over manually, but I can see how this will help with efficiency in the future. I'll give it a go and test it out.

**Pull request merged**